gnd/src/main/java/com/google/android/gnd/ui/offlinearea/selector/OfflineAreaSelectorViewModel.java

```diff
@@ -16,27 +16,51 @@
 
 package com.google.android.gnd.ui.offlinearea.selector;
 
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.LiveDataReactiveStreams;
 import com.google.android.gms.maps.model.LatLngBounds;
 import com.google.android.gnd.repository.OfflineAreaRepository;
 import com.google.android.gnd.ui.common.AbstractViewModel;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.subjects.PublishSubject;
 import javax.inject.Inject;
 import timber.log.Timber;
 
 public class OfflineAreaSelectorViewModel extends AbstractViewModel {
 
+  private final LiveData<DownloadTrigger> downloads;
+
+  public LiveData<DownloadTrigger> getDownloadTriggers() {
+    return this.downloads;
+  }
+
+  enum DownloadTrigger {
+    Started,
+    Failure
+  }
+
   private final OfflineAreaRepository offlineAreaRepository;
+  private final PublishSubject<DownloadTrigger> downloadsPublishSubject = PublishSubject.create();
 
   @Inject
   OfflineAreaSelectorViewModel(OfflineAreaRepository offlineAreaRepository) {
     this.offlineAreaRepository = offlineAreaRepository;
+    this.downloads =
+        LiveDataReactiveStreams.fromPublisher(
+            downloadsPublishSubject.toFlowable(BackpressureStrategy.LATEST));
   }
 
   // TODO: Use an abstraction over LatLngBounds
   public void onDownloadClick(LatLngBounds viewport) {
     Timber.d("viewport:%s", viewport);
-    offlineAreaRepository.addAreaAndEnqueue(viewport)
-      .doOnError(err -> Timber.e("Failed to add area and queue downloads: %s", err.getMessage()))
-      .onErrorComplete()
-      .blockingAwait();
+    offlineAreaRepository
+        .addAreaAndEnqueue(viewport)
+        .doOnError(
```

Comment:
This will report the error but it won't catch it; I believe blockingAwait() will throw an exception for unhandled errors, so may want to use try{} catch around this call instead so you'd catch both cases.

Reply:
Handled via subscribe instead